packages/opentelemetry-api/src/common/Exception.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+interface ExceptionWithType {
+  message?: string;
+  name: string;
+  stack?: string;
+}
+
+interface ExceptionWithMessage {
+  message: string;
+  name?: string;
+  stack?: string;
+}
+
+/**
+ * Defines Exception.
+ *
+ * string or an object with one of (message or name) and optional stack
+ */
+export type Exception = ExceptionWithType | ExceptionWithMessage | string;
```

Comment:
Can we also add `ExceptionWithCode`? All errors thrown by NodeJS have the `code` property now which is much more stable than name, and is a more reliable way to identify errors.

https://nodejs.org/api/errors.html#errors_error_code